Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -198,13 +202,31 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
 
   /// Returns the updated lexical tokens for the given `snapshot`.
   private func updateSyntaxTree(
-    for snapshot: DocumentSnapshot
+    for snapshot: DocumentSnapshot,
+    with edits: [IncrementalEdit]? = nil
```

Comment:
Why not make this parameter `ConcurrentEdits?` instead of `[IncrementalEdit]?`. That way the caller can decide whether the edits are sequential or concurrent.